--- driver.py	2014-07-10 13:22:21.000000000 +0000
+++ driver_new.py	2015-03-11 08:21:52.874891547 +0000
@@ -56,7 +56,6 @@
 from eventlet import greenthread
 from eventlet import patcher
 from eventlet import tpool
-from eventlet import util as eventlet_util
 from lxml import etree
 from oslo.config import cfg

@@ -622,12 +621,10 @@
         except (ImportError, NotImplementedError):
             # This is Windows compatibility -- use a socket instead
             #  of a pipe because pipes don't really exist on Windows.
-            sock = eventlet_util.__original_socket__(socket.AF_INET,
-                                                     socket.SOCK_STREAM)
+            sock = native_socket.socket(socket.AF_INET,socket.SOCK_STREAM)
             sock.bind(('localhost', 0))
             sock.listen(50)
-            csock = eventlet_util.__original_socket__(socket.AF_INET,
-                                                      socket.SOCK_STREAM)
+            csock = native_socket.socket(socket.AF_INET,socket.SOCK_STREAM)
             csock.connect(('localhost', sock.getsockname()[1]))
             nsock, addr = sock.accept()
             self._event_notify_send = nsock.makefile('wb', 0)
@@ -2448,6 +2445,8 @@
             return None

         host = CONF.serial_port_proxyclient_address
+        if host == '0.0.0.0':
+            host = utils.get_my_ipv4_address()

         # Return a descriptor for a raw TCP socket
         return {'host': host, 'port': tcp_port, 'internal_access_path': None}
         
